### PR TITLE
Add a Contributing Guide

### PR DESCRIPTION
**Pull request by settermjd**
This PR:

- adds instructions showing how to contribute to the project.

:information_desk_person: As part of the overhaul of the documentation, I want to make it as easy as possible for others to know how to contribute to the project.

### CODE REVIEW

**Comment by settermjd** [comment]
@shochdoerfer, this is still a work-in-progress, as there's a bit of extra information still required. But should be finished relatively quickly.

**Comment by shochdoerfer** [comment]
Good starting point! For the ci tasks we have some phing tasks which you should refer to in the documentation:
 - To run CodeSniffer use: ./vendor/bin/phing sniff 
 - To run PHPUnit use: ./vendor/bin/phing unit 
 - To lint the PHP code use: ./vendor/bin/phing lint

Before sending a PR one should run `composer.phar validate` as the Travis build will also run the command and will fail if the composer.lock file is not up-to-date with the changes in the composer.json file.

**Comment by settermjd** [comment]
Ok, the initial draft's now done and it's ready for review.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+When reporting issues, please provide the following information:
+
+- Component(s) affected
+- A description indicating how to reproduce the issue
+- A summary of the security vulnerability and impact
+
+We request that you contact us via the email address above and give the project contributors a chance to resolve the vulnerability and issue a new release prior to any public exposure.
+This helps protect users and provides them with a chance to upgrade and/or update in order to protect their applications.
+
+For sensitive email communications, please use [our PGP key](http://pgp.mit.edu/pks/lookup?op=vindex&search=0xEC37D631657E0A89).
```
Comment (shochdoerfer) [inline CONTRIBUTING.md]: I generated a PGP key for the opensource@bitExpert.de email address, please use [this](http://pgp.mit.edu/pks/lookup?op=vindex&search=0x6DE7F8597CECDF13) link as reference.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+
+```console
+$ git clone git@github.com:bitExpert/disco.git
+$ cd disco
+```
+
+2. Install dependencies via composer:
+
+```console
+$ curl -sS https://getcomposer.org/installer | php --
```
Comment (shochdoerfer) [inline CONTRIBUTING.md]: Not sure if it makes sense to better refer to the Composer documentation on how to [install Composer](https://getcomposer.org/download/) as their docs mention the more "secure" method.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+Delta compression using up to 2 threads.
+Compression objects: 100% (18/18), done.
+Writing objects: 100% (20/20), 8.19KiB, done.
+Total 20 (delta 12), reused 0 (delta 0)
+To ssh://git@github.com/{username}/disco.git
+   b5583aa..4f51698  HEAD -> master
+```
+### Sending a PR
+
+Before sending a PR, run `composer.phar validate`, as the Travis build will also run the command and will fail if the composer.lock file is not up-to-date with the changes in the composer.json file.
```
Comment (shochdoerfer) [inline CONTRIBUTING.md]: It should be mentioned that the CodeSniffer tests as well the PHPUnit tests need also to run.

**Review by shochdoerfer — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 2 above)
Comment (settermjd) [reply CONTRIBUTING.md]: Sounds good to me.

File: CONTRIBUTING.md (hunk 3 above)
Comment (settermjd) [reply CONTRIBUTING.md]: Shall do. Thanks for the pointers while I'm still getting up to speed with the project.

**Comment by shochdoerfer** [comment]
@settermjd Could you apply the few changes I mentioned above so that I can merge the PR?

**Comment by settermjd** [comment]
@shochdoerfer done!

**Comment by shochdoerfer** [comment]
Thank you @settermjd for your first Disco PR!

**Comment by settermjd** [comment]
Not a problem at all. Here's to many more!

**Pull request merged**